applications/nfc/helpers/nfc_generators.c

```diff
@@ -268,70 +314,188 @@ static void nfc_generate_ntag_i2c_plus_2k(NfcDeviceData* data) {
     mful->version.storage_size = 0x15;
 }
 
+static void nfc_generate_mf_classic_1k(NfcDeviceData* data, uint8_t uid_len) {
+    nfc_generate_common_start(data);
+    nfc_generate_mf_classic_common(data, uid_len, MfClassicType1k);
+
+    // Set the UID
+    data->nfc_data.uid[0] = NXP_MANUFACTURER_ID;
+    for(int i = 1; i < uid_len; i++) {
+        data->nfc_data.uid[i] = data->mf_classic_data.block[0].value[i];
+    }
+
+    MfClassicData* mfc = &data->mf_classic_data;
+    mf_classic_set_block_read(mfc, 0, &mfc->block[0]);
+
+    // Set every block to 0xFF
+    for(uint16_t i = 1; i < MF_CLASSIC_1K_TOTAL_SECTORS_NUM * 4; i += 1) {
+        memset(&mfc->block[i].value, 0xFF, 16);
+        mf_classic_set_block_read(mfc, i, &mfc->block[i]);
```

Comment:
Next for cycle is not necessary
if(mf_classic_is_sector_trailer(i)) {
    nfc_generate_mf_classic_sector_trailer(mfc, i);
 } else {
    memset(&mfc->block[i].value, 0xFF, 16);
    }
mf_classic_set_block_read(mfc, i, &mfc->block[i]);